### PR TITLE
DOCS-5856 integrated Community and Exchange search

### PR DESCRIPTION
**Pull request by wwarner-salesforce**
DOCS-5856 integrated Community and Exchange search

### CODE REVIEW

**Review by cominguez — COMMENTED** [review]
@diegor do you want to review this please?

File: modules/ROOT/pages/acm-lightning-components.adoc [hunk 1]
```diff
@@ -269,6 +269,43 @@ Card Shadow:: Specifies the shadow beneath the card.
...
+
+Search is a basic expected feature in any developer portal, and becomes important as the number of APIs grows to around 50 and beyond.
+
+The example portals and defaults have one Integrated Search Input and one Integrated Search Response Page. This is the best practice to minimize the chance of misconfigurations.
+
+The Integrated Search Input parameter Linked Result Page specifies the search results page. That page must have an Integrated Search Response Page component to display the search results.
+
+The Integrated Search Response Page component shows only results of the content types specified in the Search Areas parameters of the Integrated Search Input component. If a content type is not selected then the search results page does not display any search results of that content type.
+
+The current version of the integrated search does not include CMS articles in the search results.
```
Comment (paigehargrave) [inline modules/ROOT/pages/acm-lightning-components.adoc]: This needs to be called out with `IMPORTANT`.

Comment (wwarner-salesforce) [reply modules/ROOT/pages/acm-lightning-components.adoc]: Done. Thank you!

File: modules/ROOT/pages/acm-lightning-components.adoc [hunk 2]
```diff
@@ -269,6 +269,43 @@ Card Shadow:: Specifies the shadow beneath the card.
 Card Width:: Specifies the card width.
 Card Height:: Specifies the card height.
 
+== ACM Search Box
+
+The ACM Search Box component enables users to search data integrated from both Community Cloud and Anypoint Platform.
```
Comment (jdeoliveira) [inline modules/ROOT/pages/acm-lightning-components.adoc]: ... enables users to search for content that exists in both Community Cloud, like articles, announcements and forum posts, and the Anypoint Platform, like APIs and client applications. Results from the search are presented in the ACM Search Results component [ADD LINK]

File: modules/ROOT/pages/acm-lightning-components.adoc [hunk 3]
```diff
@@ -269,6 +269,43 @@ Card Shadow:: Specifies the shadow beneath the card.
...
+
+IMPORTANT: The current version of the integrated search does not include CMS articles in the search results.
+
+Search results are sorted by the date they were most recently modified, from the most recent date to the oldest.
+
+To test and preview the search configuration and the presentation of the results page, xref:preview-community.adoc[Preview Your API Community] and run a search.
+
+== ACM Search Results
+
+The ACM Search Results component shows results after a user enters a search.
```
Comment (jdeoliveira) [inline modules/ROOT/pages/acm-lightning-components.adoc]: ... a user enters a search in the ACM Search Box component. [ADD LINK]

**Review by jdeoliveira — CHANGES_REQUESTED** [review]

File: modules/ROOT/pages/acm-lightning-components.adoc (hunk 2 above)
Comment (wwarner-salesforce) [reply modules/ROOT/pages/acm-lightning-components.adoc]: Made this change. Much better! Thank you.

File: modules/ROOT/pages/acm-lightning-components.adoc (hunk 3 above)
Comment (wwarner-salesforce) [reply modules/ROOT/pages/acm-lightning-components.adoc]: Made this change. Thanks!

**Review by jdeoliveira — APPROVED** [review]

File: modules/ROOT/pages/acm-lightning-components.adoc [hunk 4]
```diff
@@ -12,6 +12,43 @@ Each ACM Lightning component has different configuration options and adds differ
 
 Many ACM Lightning components show data from Anypoint Platform. ACM admins can change some of these data fields, such as API names, descriptions, and icons. ACM saves these changes but does not push the changes back to Anypoint Platform.
 
+== ACM Search Box [[acm-search-box]]
+
+The ACM Search Box component enables users to search both for content that exists in Community Cloud, such as knowledge articles, support cases, and announcements, and content that exists in Anypoint Platform, such as APIs and client applications. Results from the search are presented in the <<acm-search-results>> component.
```
Comment (kbostrom) [inline modules/ROOT/pages/acm-lightning-components.adoc]:
```suggestion
The Anypoint API Community Manager Search Box component enables users to search for both content that exists in Community Cloud (such as knowledge articles, support cases, and announcements) and content that exists in Anypoint Platform (such as APIs and client applications). Results from the search are presented in the <<acm-search-results>> component.
```

File: modules/ROOT/pages/acm-lightning-components.adoc [hunk 5]
```diff
@@ -12,6 +12,43 @@ Each ACM Lightning component has different configuration options and adds differ
...
+
+You can specify values for the following parameters of the ACM Search Box:
+
+Component Title:: Specifies the title for the search input field.
+Number of Results - Instant Incremental Response:: Specifies the number of results shown as the user types a query.
+Search Areas:: Selects which types of content to search. Choices include APIs, Applications, Knowledge Articles, Cases, Pages, and Forums.
+Linked Result Page:: Specifies the URL of the search results page shown after the user enters a search.
+
+=== Search Configuration
```
Comment (kbostrom) [inline modules/ROOT/pages/acm-lightning-components.adoc]: Is this solitary subtopic necessary, as a subtopic?
You typically want at least two before you break a topic into subs....
Perhaps the container topic should have a title that incorporates this info?
Or...is this actually a mislabeled Level-2 peer topic? These are all components?

File: modules/ROOT/pages/acm-lightning-components.adoc [hunk 6]
```diff
@@ -12,6 +12,43 @@ Each ACM Lightning component has different configuration options and adds differ
...
+
+Search is a basic expected feature in any developer portal, and becomes important as the number of APIs grows to around 50 and beyond.
```
Comment (kbostrom) [inline modules/ROOT/pages/acm-lightning-components.adoc]: This intro sentence sounds gratuitous to me. We all know about Search and why it is important for large amounts of data....

File: modules/ROOT/pages/acm-lightning-components.adoc [hunk 7]
```diff
@@ -12,6 +12,43 @@ Each ACM Lightning component has different configuration options and adds differ
...
+
+The example portals and default pages have one ACM Search Box and one ACM Search Results component. This is the best practice to minimize the chance of misconfigurations.
```
Comment (kbostrom) [inline modules/ROOT/pages/acm-lightning-components.adoc]: The example portals and default pages of what? Some kind of template?
Is this info necessary here?
Why would there be more than one search or results component?

File: modules/ROOT/pages/acm-lightning-components.adoc [hunk 8]
```diff
@@ -12,6 +12,43 @@ Each ACM Lightning component has different configuration options and adds differ
...
+
+The ACM Search Box parameter Linked Result Page specifies the search results page. That page must have an ACM Search Results component to display the search results.
```
Comment (kbostrom) [inline modules/ROOT/pages/acm-lightning-components.adoc]: Should this sentence be up above, where the parameters are described?
And...can you reread these two sentences? I can't make any sense of them, but maybe I am having some sort of block....

File: modules/ROOT/pages/acm-lightning-components.adoc [hunk 9]
```diff
@@ -12,6 +12,43 @@ Each ACM Lightning component has different configuration options and adds differ
...
+
+The ACM Search Results component shows only results of the content types specified in the Search Areas parameters of the ACM Search Box component. If a content type is not selected then the search results do not include any results of that content type.
```
Comment (kbostrom) [inline modules/ROOT/pages/acm-lightning-components.adoc]: Shouldn't this text be moved to the following subtopic by the same name?
Also, GLOBALLY fully spell or use the accepted short form API Community Manager rather than ACM.
```suggestion
The API Community Manager Search Results component shows results of only the content types specified in the Search Areas parameters of the API Community Manager Search Box component. If a content type is not selected then the search results do not include any results of that content type.
```

File: modules/ROOT/pages/acm-lightning-components.adoc [hunk 10]
```diff
@@ -12,6 +12,43 @@ Each ACM Lightning component has different configuration options and adds differ
...
+
+IMPORTANT: The current version of the integrated search does not include CMS articles in the search results.
```
Comment (kbostrom) [inline modules/ROOT/pages/acm-lightning-components.adoc]: Can we specify the version number that we mean by "current," to insure against mistakes during updates?

File: modules/ROOT/pages/acm-lightning-components.adoc [hunk 11]
```diff
@@ -12,6 +12,43 @@ Each ACM Lightning component has different configuration options and adds differ
...
+
+Search results are sorted by the date they were most recently modified, from the most recent date to the oldest.
```
Comment (kbostrom) [inline modules/ROOT/pages/acm-lightning-components.adoc]:
```suggestion
Search results are sorted by the date they were most recently modified, from newest to oldest.
```

File: modules/ROOT/pages/acm-lightning-components.adoc [hunk 12]
```diff
@@ -12,6 +12,43 @@ Each ACM Lightning component has different configuration options and adds differ
...
+
+To test and preview the search configuration and the presentation of the results page, xref:preview-community.adoc[Preview Your API Community] and run a search.
```
Comment (kbostrom) [inline modules/ROOT/pages/acm-lightning-components.adoc]:
```suggestion
To test and preview the search configuration and the presentation of the results page, xref:preview-community.adoc[preview your API community] and run a search.
```

File: modules/ROOT/pages/acm-lightning-components.adoc [hunk 13]
```diff
@@ -12,6 +12,43 @@ Each ACM Lightning component has different configuration options and adds differ
 
 Many ACM Lightning components show data from Anypoint Platform. ACM admins can change some of these data fields, such as API names, descriptions, and icons. ACM saves these changes but does not push the changes back to Anypoint Platform.
 
+== ACM Search Box [[acm-search-box]]
+
+The ACM Search Box component enables users to search both for content that exists in Community Cloud, such as knowledge articles, support cases, and announcements, and content that exists in Anypoint Platform, such as APIs and client applications. Results from the search are presented in the <<acm-search-results>> component.
+
+You can specify values for the following parameters of the ACM Search Box:
```
Comment (kbostrom) [inline modules/ROOT/pages/acm-lightning-components.adoc]:
```suggestion
You can specify values for the following parameters of the API Community Manager Search Box:
```

File: modules/ROOT/pages/acm-lightning-components.adoc [hunk 14]
```diff
@@ -12,6 +12,43 @@ Each ACM Lightning component has different configuration options and adds differ
...
+
+IMPORTANT: The current version of the integrated search does not include CMS articles in the search results.
+
+Search results are sorted by the date they were most recently modified, from the most recent date to the oldest.
+
+To test and preview the search configuration and the presentation of the results page, xref:preview-community.adoc[Preview Your API Community] and run a search.
+
+== ACM Search Results [[acm-search-results]]
+
+The ACM Search Results component shows results after a user enters a search in the <<acm-search-box>> component.
```
Comment (kbostrom) [inline modules/ROOT/pages/acm-lightning-components.adoc]:
```suggestion
The API Community Manager Search Results component shows results after a user enters a search in the <<acm-search-box>> component.
```

File: modules/ROOT/pages/acm-lightning-components.adoc [hunk 15]
```diff
@@ -12,6 +12,43 @@ Each ACM Lightning component has different configuration options and adds differ
...
+
+You can specify values for the following parameters of the ACM Search Results:
```
Comment (kbostrom) [inline modules/ROOT/pages/acm-lightning-components.adoc]:
```suggestion
You can specify values for the following parameters of the API Community Manager Search Results:
```
OR
```suggestion
You can specify values for the following parameters of the Search Results:
```

**Review by kbostrom — APPROVED** [review]
I found this a bit hard to follow, in terms of narrative flow. Please carefully reread this yourself while evaluating my suggestions.

File: modules/ROOT/pages/acm-lightning-components.adoc (hunk 5 above)
Comment (wwarner-salesforce) [reply modules/ROOT/pages/acm-lightning-components.adoc]: Good point. The only two components are "Search Box" and "Search Results", and this configuration info applies to the "Search Box", so it's better without a separate heading. Removed it.

File: modules/ROOT/pages/acm-lightning-components.adoc (hunk 6 above)
Comment (wwarner-salesforce) [reply modules/ROOT/pages/acm-lightning-components.adoc]: True, better to omit this. Removed.

File: modules/ROOT/pages/acm-lightning-components.adoc (hunk 7 above)
Comment (wwarner-salesforce) [reply modules/ROOT/pages/acm-lightning-components.adoc]: API Community Manager comes with a default setup, and community admins can add, remove, and move pages and components to rebuild the community site in any way. Admins may wish to separate site members into audiences, such as internal developers at their organization and external developers at other organizations, and each audience could have their own search, which could automatically filter out APIs that would only be used by the other group. However, in most cases a community will just have one search component and one results component. I've rewritten this part for better clarity.

File: modules/ROOT/pages/acm-lightning-components.adoc (hunk 8 above)
Comment (wwarner-salesforce) [reply modules/ROOT/pages/acm-lightning-components.adoc]: Yes, better to move those up. I also rewrote for clarity: "The Linked Result Page parameter specifies the URL of a search results page, and the search results page must include an ACM Search Results component to display the search results. If the page does not include an ACM Search Results component, members and guest users who enter a search will go to the page and see no results."

File: modules/ROOT/pages/acm-lightning-components.adoc (hunk 9 above)
Comment (wwarner-salesforce) [reply modules/ROOT/pages/acm-lightning-components.adoc]: Rewrote this to clarify that it's focused on the Search Areas parameter.

Unfortunately the UI has a long alphabetical list of components, including the names "ACM Search Box" and "ACM Search Results", so if the docs call them "API Community Manager Search Box" and so on, we'll send the users to the wrong place in the list. I'll raise this with the PM next week when I scrub "ACM" from all pages in the docs, and hopefully we can get it fixed in the UI.

File: modules/ROOT/pages/acm-lightning-components.adoc (hunk 10 above)
Comment (wwarner-salesforce) [reply modules/ROOT/pages/acm-lightning-components.adoc]: Yes, that's much better. Rewrote and will add to the final release to-do list to add the package version here.
